digit-health-services/health-sync-service/src/main/java/org/digit/health/sync/repository/QueryFieldConditionChecker.java

```diff
@@ -0,0 +1,11 @@
+package org.digit.health.sync.repository;
+
+import java.lang.reflect.Field;
+import java.util.Optional;
+
+public interface QueryFieldConditionChecker {
+    boolean check(Field field, Object object) throws IllegalAccessException;
+
+    QueryFieldConditionChecker checkIfFieldIsNotNull = (field, object) -> Optional.ofNullable(field.get(object)).isPresent();
```

Comment:
Remove checkIf
Rename to isNotNull
Rename to isAnnotatedWithUpdateBy

Reply:
Done.